crates/napi/src/threadsafe_function.rs

```diff
@@ -341,6 +350,11 @@ unsafe extern "C" fn call_js_cb<T: 'static, V: NapiRaw, R, ES>(
   R: 'static + Send + FnMut(ThreadSafeCallContext<T>) -> Result<Vec<V>>,
   ES: ErrorStrategy::T,
 {
+  // env and/or callback can be null when shutting down
+  if raw_env.is_null() || js_callback.is_null() {
+    return;
+  }
```

Comment:
When V8 cleans up, it clears the call queue by calling the registered callback (`call_js_cb`) with null values for the `env` and `js_callback` args. Added a check to return early in this case.